File: backend/app/models/menu.rb
class Menu < ApplicationRecord
  validates :name, presence: true
  validates :price, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
  validates :category, presence: true
  validates :description, presence: true, length: { maximum: 255 }
end

File: admin_frontend/types/menu.ts
export interface Menu {
  id: number
  name: string
  price: number
  category: string
  description?: string
  is_published: boolean
  created_at?: string
  updated_at?: string
}

export interface MenuFormData {
  name: string
  price: number | string
  category: string
  description?: string
  is_published: boolean
}
